### PR TITLE
NF: Split TagsDialog into multiple files 

### PR DESCRIPTION
**Pull request by TarekkMA**
## Purpose / Description
This is a continuation of #8322, moved to its own PR to reduce review complexity of the original PR

## Fixes
Fixes N/A

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsList.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
+package com.ichi2.anki.dialogs.tags;
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsList.java:1]: I checked this move with a diff tool, and it's okay

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package com.ichi2.anki.dialogs.tags;
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java:1]: I checked this move with a diff tool and it's okay

**Review by Arthur-Milchior — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java (hunk 2 above)
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java:1]: missing copyright header @TarekkMA

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsList.java (hunk 1 above)
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsList.java:1]: missing copyright header

**Review by mikehardy — CHANGES_REQUESTED** [review]
2 missing copyright header notes but otherwise yes looks NF and should be fine?
Add the copyright headers and then anyone with merge power that sees this please merge so we may unblock #8577 :-)

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java (hunk 2 above)
Comment (TarekkMA) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.java:1]: alright I will add it

**Comment by TarekkMA** [comment]
I did add the copyright headers.

**Comment by mikehardy** [comment]
#8577 unblocked! Yesss! :-)

**Comment by mikehardy** [comment]
Hi there!

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for April 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged in April)

Thanks!

**Pull request merged**